### PR TITLE
add metaflow handling to channels

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/crossbar/doc/channels.md [hunk 1]
```diff
@@ -111,3 +111,14 @@ curl -v -X POST \
     -d '{"data": {"action": "transfer", "target": "2600", "takeback_dtmf": "*1", "moh": "media_id" }}' \
     http://{SERVER}:8000/v2/accounts/{ACCOUNT_ID}/channels/{UUID}
 ```
+
+#### Put a feature on a channel
```
Comment (jamesaimonetti) [inline applications/crossbar/doc/channels.md]: Can you add a small description? For instance, what types of features are available?

Also, can you add example JSON payloads for the request?

File: applications/crossbar/src/modules/cb_channels.erl [hunk 2]
```diff
@@ -125,6 +129,24 @@ post(Context, _UUID) ->
...
+%% (after a merge perhaps).
+%% @end
+%%--------------------------------------------------------------------
+-spec put(cb_context:context(), path_token()) -> cb_context:context().
+put(Context, UUID) ->
+    API = [{<<"Call-ID">>, UUID}
+          ,{<<"Flow">>, cb_context:doc(Context)}
+           | kz_api:default_headers(?APP_NAME, ?APP_VERSION)
+          ],
+    lager:debug("SENDING ACTION ~p", [API]),
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_channels.erl]: Why are you shouting?!? Maybe remove this?

File: applications/crossbar/src/modules/cb_channels.erl [hunk 3]
```diff
@@ -212,6 +234,25 @@ maybe_execute_command(Context, _CallId, _Command) ->
...
+-spec validate_action(cb_context:context(), ne_binary(), api_binary()) -> cb_context:context().
+validate_action(Context, CallId) ->
+    Ctx = read(Context, CallId),
+    case cb_context:has_errors(Ctx) of
+        'true' -> Ctx;
+        'false' -> validate_action(Ctx, CallId, cb_context:req_value(Context, <<"action">>))
+    end.
+
+validate_action(Context, _UUID, <<"metaflow">>) ->
+    lager:debug("VALIDATING"),
```
Comment (jamesaimonetti) [inline applications/crossbar/src/modules/cb_channels.erl]: LOUD NOISES! Maybe remove?

File: applications/ecallmgr/src/ecallmgr_call_command.erl [hunk 4]
```diff
@@ -1620,3 +1626,37 @@ transfer_leg(JObj) ->
...
 transfer_context(JObj) ->
     kz_json:get_value(<<"Transfer-Context">>, JObj, ?DEFAULT_FREESWITCH_CONTEXT).
+
+-spec sound_touch(ne_binary(), ne_binary(), kz_json:object()) -> {ne_binary(), ne_binary()}.
+sound_touch(UUID, <<"start">>, JObj) ->
+    {<<"soundtouch">>, list_to_binary([UUID, " start ", sound_touch_options(JObj)])};
+sound_touch(UUID, <<"stop">>, _JObj) ->
+    {<<"soundtouch">>, list_to_binary([UUID, " stop"])}.
+
+sound_touch_options(JObj) ->
```
Comment (jamesaimonetti) [inline applications/ecallmgr/src/ecallmgr_call_command.erl]: spec needed

File: applications/crossbar/doc/channels.md (hunk 1 above)
Comment (lazedo) [inline applications/crossbar/doc/channels.md]: done

File: applications/crossbar/src/modules/cb_channels.erl (hunk 2 above)
Comment (lazedo) [inline applications/crossbar/src/modules/cb_channels.erl]: done

File: applications/crossbar/src/modules/cb_channels.erl (hunk 3 above)
Comment (lazedo) [inline applications/crossbar/src/modules/cb_channels.erl]: done

File: applications/ecallmgr/src/ecallmgr_call_command.erl (hunk 4 above)
Comment (lazedo) [inline applications/ecallmgr/src/ecallmgr_call_command.erl]: done

**Pull request merged**